Build fails with CMake version < 3.14

This error is caused by using the option `ARCH_INDEPENDENT`, which was only added in CMake version 3.14 (see the release notes for that version [here](https://cmake.org/cmake/help/latest/release/3.14.html?highlight=arch_independent#modules)). Building the library fails with older versions of CMake. For example, with CMake 3.11 I get:

```
CMake Error at /usr/share/cmake/Modules/WriteBasicConfigVersionFile.cmake:30 (message):
  Unknown keywords given to WRITE_BASIC_CONFIG_VERSION_FILE():
  "ARCH_INDEPENDENT"
Call Stack (most recent call first):
  /usr/share/cmake/Modules/CMakePackageConfigHelpers.cmake:209 (write_basic_config_version_file)
  CMakeLists.txt:116 (write_basic_package_version_file)
```

[This recent commit](https://github.com/pybind/pybind11_json/commit/1baeddbf9e44ac44d11e0a6e5e12eb32068c6e28) introduced the use of `ARCH_INDEPENDENT`, but did not change the minimum required CMake version.

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -6,7 +6,7 @@
 # The full license is in the file LICENSE, distributed with this software. #
 ############################################################################
 
-cmake_minimum_required(VERSION 3.4.3)
+cmake_minimum_required(VERSION 3.14)
 project(pybind11_json)
 
 set(PYBIND11_JSON_INCLUDE_DIR ${CMAKE_CURRENT_SOURCE_DIR}/include)

